openssl/aes.go

```diff
@@ -178,37 +146,29 @@ func (x *aesCBC) SetIV(iv []byte) {
 	if len(iv) != aesBlockSize {
 		panic("cipher: incorrect length IV")
 	}
-	copy(x.iv[:], iv)
-	if C.int(1) != C._goboringcrypto_EVP_CipherInit_ex(x.ctx, nil, nil, nil, (*C.uchar)(unsafe.Pointer(&x.iv[0])), -1) {
+	if C.go_openssl_EVP_CipherInit_ex(x.ctx, nil, nil, nil, base(iv), -1) != 1 {
 		panic("cipher: unable to initialize EVP cipher ctx")
 	}
 }
 
 func (c *aesCipher) NewCBCEncrypter(iv []byte) cipher.BlockMode {
-	x := &aesCBC{key: c.key, mode: C.GO_AES_ENCRYPT}
-	copy(x.iv[:], iv)
```

Comment:
I'm not copying `iv` here because `EVP_CipherInit_ex` already copies it before storing.